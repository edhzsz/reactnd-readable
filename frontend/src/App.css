/* palette to use: https://material.io/color/#!/?view.left=0&view.right=0&secondary.color=BBDEFB&primary.color=546E7A */

.app {
  color: #fff;
  height: 100%;
}

/* Layout */
.main {
  padding-top: 64px;
  height: calc(100% - 128px);
  background-color: #546e7a;
}

.content {
  padding: 38px 42px;
  overflow: auto;
  float: left;
  width: calc(100% - 340px);
  height: calc(100% - 100px);
}

/* Header and Footer Bar */
.header {
  background-color: #29434e;
  box-sizing: border-box;
  border-radius: 0px;
  position: fixed;
  z-index: 1101;
  width: 100%;
  display: flex;
  padding-left: 24px;
  padding-right: 24px;
}

.header>h1 {
  white-space: nowrap;
  overflow: hidden;
  margin: 0px;
  padding-top: 0px;
  letter-spacing: 0px;
  font-size: 24px;
  font-weight: 400;
  height: 64px;
  line-height: 64px;
  flex: 1 1 0%;
}

.footer {
  background-color: #819ca9;
  box-sizing: border-box;
  border-radius: 0px;
  position: fixed;
  z-index: 1101;
  top: calc(100% - 64px);
  width: 100%;
  display: flex;
  padding-left: 24px;
  padding-right: 24px;
  height: 64px;
}

.side-bar {
  box-sizing: border-box;
  font-family: Roboto, sans-serif;
  box-shadow: rgba(0, 0, 0, 0.16) 0px 3px 10px, rgba(0, 0, 0, 0.23) 0px 3px 10px;
  border-radius: 0px;
  height: 100%;
  width: 256px;
  z-index: 1200;
  left: 0px;
  top: 0px;
  transform: translate(0px, 0px);
  overflow: auto;
  float: left;
  background-color: #F5F5F6;
  color: #000;
}

.title {
  font-size: 24px;
  line-height: 64px;
  font-weight: 300;
  padding: 0;
  padding-left: 24px;
  margin: 0;
  margin-bottom: 8px;
  color: #000;
}

/* Buttons */
.btn-rect {
  border: 10px;
  box-sizing: border-box;
  display: inline-block;
  cursor: pointer;
  text-decoration: none;
  margin: 0px;
  padding: 0px;
  outline: none;
  position: relative;
  z-index: 1;
  height: 36px;
  line-height: 36px;
  border-radius: 2px;
  text-align: center;
}

.btn-rect span {
  opacity: 1;
  font-size: 14px;
  letter-spacing: 0px;
  text-transform: uppercase;
  font-weight: 500;
  margin: 0px;
  user-select: none;
  padding-left: 16px;
  padding-right: 16px;
}

.base {
  background-color: #819ca9;
}

.primary {
  color: #000;
  background-color: #89abc6;
}

.secondary {
  color: #000;
  background-color: #baddf9;
}
