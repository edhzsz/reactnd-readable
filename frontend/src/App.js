import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="app">
        <div className="navbar">
          <h1>Readable</h1>
        </div>
        <div className="main">
          <div className="content">
            Hello World!
            <button type="button" className="btn-rect primary">
              <span>Primary</span>
            </button>
            <button type="button" className="btn-rect secondary">
              <span>Secondary</span>
            </button>
          </div>
        </div>
        <div className="footer">
          
        </div>
      </div>
    );
  }
}

export default App;
