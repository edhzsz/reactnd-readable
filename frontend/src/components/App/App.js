import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { connect } from 'react-redux';

export class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          Hello { this.props.name }!
        </p>
      </div>
    );
  }
}

export function mapStateToProps(state) {
  return {
    name: 'Edgar'
  }
}

export default connect(mapStateToProps)(App);
