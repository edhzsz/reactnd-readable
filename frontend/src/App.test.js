import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

const wrapper = shallow(<App/>);

describe('Component: App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('should match its empty snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});