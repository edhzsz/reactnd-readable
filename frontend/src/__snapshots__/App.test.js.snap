// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should match its empty snapshot 1`] = `
<div
  className="app"
>
  <div
    className="header"
  >
    <h1>
      Readable
    </h1>
  </div>
  <div
    className="main"
  >
    <div
      className="side-bar"
    >
      <h2
        className="title base"
      >
        Categories
      </h2>
      <ul>
        <li>
          All
        </li>
        <li>
          React
        </li>
        <li>
          Redux
        </li>
        <li>
          Udacity
        </li>
        <li>
          + Add Category
        </li>
      </ul>
    </div>
    <div
      className="content"
    >
      <div
        className="post-list"
      >
        <div
          className="post"
        >
          <div
            className="post-header base"
          >
            <span>
              Udacity is the best place to learn React
            </span>
          </div>
          <div
            className="post-contents"
          >
            <div
              className="post-votes"
            >
              <RoundButton
                backgroundImage="down_arrow.svg"
                className="up-vote"
                size={30}
                type="secondary"
              >
                Up Vote
              </RoundButton>
              <span>
                6
              </span>
              <RoundButton
                backgroundImage="down_arrow.svg"
                size={30}
                type="secondary"
              >
                Down Vote
              </RoundButton>
            </div>
            <div
              className="post-wrapper"
            >
              <div
                className="post-main"
              >
                Everyone says so after all.
              </div>
              <div
                className="comments-list"
              >
                <span
                  className="title"
                >
                  Comments
                </span>
                <ul>
                  <li>
                    <div>
                      <span
                        className="comment-content"
                      >
                        Hi there! I am a COMMENT.
                      </span>
                      <span
                        className="comment-author"
                      >
                        thingtwo
                      </span>
                    </div>
                  </li>
                  <li>
                    <div>
                      <span
                        className="comment-content"
                      >
                        Comments. Are. Cool.
                      </span>
                      <span
                        className="comment-author"
                      >
                        thingone
                      </span>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <RoundButton
      backgroundImage="add.svg"
      className="add-post"
      size={50}
      type="primary"
    >
      Add a Book
    </RoundButton>
  </div>
  <div
    className="footer"
  />
</div>
`;
